docs/user-guide.md

```diff
@@ -90,13 +88,42 @@ be too. You might use this method like this:
 Tools could then use the `@Nullable` information to determine that the first use
 is safe but the second is not.
 
+As far as JSpecify is concerned, `String` and `@Nullable String` are _different_
+types. A variable of type `String` can reference any string. A variable
+of type `@Nullable String` can too, but it can also be null. This means that
+`String` is a _subtype_ of `@Nullable String`, in the same way that `Integer` is
+a subtype of `Number`. One way to look at this is that a subtype narrows the
+range of possibilities. A `Number` variable can be assigned from an `Integer`
+but it can also be assigned from a `Long`. Meanwhile an `Integer` variable
+can't be assigned from a `Number` (since that `Number` might be a `Long` or some
+other subtype). In the same way, a `@Nullable String` can be assigned from a
+`String` but a `String` can't be assigned from a `@Nullable String` (since that
+might be null).
```

Comment:
Can this be simplified to say that a subtype contains fewer values than a supertype? And the examples can refer to actual values, like "some Numbers are not Integers, such as 3.2".

Although maybe a domain that isn't boxed primitives would be better?

Reply:
This is tough. I'm not particularly happy with the current text, but I want to avoid making it harder to understand. The example of `Number` with subtypes `Integer` and `Long` is very familiar and I'm not sure there's a ton of other examples that would be equally familiar. We could say that `String` and `StringBuilder` are both subtypes of `CharSequence` but I think that would confusing, because then we're talking about `String` being a subtype of two different things (`CharSequence` and `@Nullable String`) in two different ways (in the language vs in JSpecify).

I don't want to get into boxing, so I don't want to say `3.2`, any more than I wanted to say `"a"` as an example of something that a `String` variable can reference. These are things that are treated specially by the language (string literals and autoboxing) and I want to make it clear that exactly the same considerations would apply to `Object` or `Thread`, for example.

Maybe a picture would help, either instead of or as well as all this text?

Reply:
Nah, I think what you have here is okay, at least for now.